### PR TITLE
POEM 74: Suggest variables for failed connection

### PR DESCRIPTION
**Pull request by joanibal**
### Summary
This PR is associated with POEM 74: Suggest variables close in name on failed connection attempt. 

### Related Issues

None

### Backwards incompatibilities

None. It only changes error messages

### New Dependencies

The python built-in difflib

### CODE REVIEW

File: openmdao/core/group.py [hunk 1]
```diff
@@ -1459,8 +1460,10 @@ def _setup_global_connections(self, conns=None):
                                       err=self._raise_connection_errors)
                     continue
                 else:
+                    guesses = get_close_matches(prom_out, list(allprocs_prom2abs_list_out.keys()) + \
+                                                         list(allprocs_discrete_out.keys()))
                     msg = f"{self.msginfo}: Attempted to connect from '{prom_out}' to " + \
-                          f"'{prom_in}', but '{prom_out}' doesn't exist."
+                          f"'{prom_in}', but '{prom_out}' doesn't exist.Perhaps you ment to connect to one of the following outputs:{guesses}"
```
Comment (naylor-b) [inline openmdao/core/group.py:1466]: ment --> meant (also occurs below)

**Review by naylor-b — CHANGES_REQUESTED** [review]

**Review by robfalck — CHANGES_REQUESTED** [review]

**Comment by joanibal** [comment]
Thanks for the grammar fixes @johnjasa

**Comment by joanibal** [comment]
It appears I can only add a re-review request to one reviewer

**Review by naylor-b — APPROVED** [review]

**Comment by johnjasa** [comment]
@joanibal will you be able to address these test failures? I can take a stab if you prefer. I think it's just pep8 and an error message test for your updated message.

**Comment by joanibal** [comment]
The tests now seem to be failing for reasons unrelated to the PR.
Did something in the CI pipeline change?

On Ubuntu_minmal the error is related to the `open` function (possibly related to the python 3.11 update?)
```
   File "/usr/share/miniconda/envs/test/lib/python3.11/site-packages/openmdao/devtools/iprof_utils.py", line 80, in find_qualified_name
    with open(filename, 'Ur') as f:
         ^^^^^^^^^^^^^^^^^^^^
ValueError: invalid mode: 'Ur'
```

The other platforms are failing the dependency audit 
```
Found 1 known vulnerability, ignored 2 in 1 package
Name         Version ID                  Fix Versions
------------ ------- ------------------- ------------
jupyter-core 4.11.1  GHSA-m678-f26j-3hrp 4.11.2
```
or can't pip install jax
```
Collecting jax
  Downloading jax-0.3.23.tar.gz (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 27.4 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
ERROR: Could not find a version that satisfies the requirement jaxlib (from versions: none)
ERROR: No matching distribution found for jaxlib
```

**Comment by swryan** [comment]
@joanibal  you should be good to sync your PRs up with the master branch now..  the GitHub workflow has been updated.

Thanks for your patience.

**Review by robfalck — APPROVED** [review]

**Pull request merged**